apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java

```diff
@@ -1503,4 +1828,156 @@ private void writeDebugLogMessages() {
+        } else {
+          /*
+           * Unfortunately, Hibernate's SQL schema generation (HBM2DDL) doesn't correctly
+           * handle SQL numeric datatypes that don't have a defined precision and scale.
+           * What it _should_ do is represent those types in PostgreSQL as a "NUMERIC",
+           * but what it does instead is insert a default precision and scale as
+           * "NUMBER(19, 2)". The only way to force the correct behavior is to specify a
+           * columnDefinition, so we do that. This leads to incorrect behavior with HSQL
+           * (for different reasons), but fortunately that doesn't happen to cause
+           * problems with our tests.
+           */
+          StringBuilder columnDefinition = new StringBuilder();
+          columnDefinition.append("numeric");
+          if (columnLength.isPresent() || columnScale.isPresent()) {
+            columnDefinition.append('(');
+            if (columnLength.isPresent()) {
+              columnDefinition.append(columnLength.get());
+            }
+            if (columnScale.isPresent()) {
+              columnDefinition.append(", ");
+              columnDefinition.append(columnScale.get());
+            }
+            columnDefinition.append(')');
+          }
+          columnAnnotation.addMember("columnDefinition", "$S", columnDefinition.toString());
+        }
+      }
+      annotations.add(columnAnnotation.build());
+    } else {
+      annotations.add(AnnotationSpec.builder(Transient.class).build());
+    }
+
+    return annotations;
+  }
+
+  private static TypeName selectJavaFieldType(
+      RifColumnType type,
+      boolean isColumnOptional,
+      Optional<Integer> columnLength,
+      Optional<Integer> columnScale) {
+    if (type == RifColumnType.CHAR
+        && columnLength.orElse(Integer.MAX_VALUE) == 1
+        && !isColumnOptional) return TypeName.CHAR;
+    else if (type == RifColumnType.CHAR
+        && columnLength.orElse(Integer.MAX_VALUE) == 1
+        && isColumnOptional) return ClassName.get(Character.class);
+    else if (type == RifColumnType.CHAR) return ClassName.get(String.class);
+    else if (type == RifColumnType.DATE && columnLength.orElse(0) == 8)
+      return ClassName.get(LocalDate.class);
+    else if (type == RifColumnType.TIMESTAMP && columnLength.orElse(0) == 20)
+      return ClassName.get(Instant.class);
+    else if (type == RifColumnType.NUM && columnScale.orElse(Integer.MAX_VALUE) > 0)
+      return ClassName.get(BigDecimal.class);
+    else if (type == RifColumnType.NUM
+        && columnScale.orElse(Integer.MAX_VALUE) == 0
+        && !isColumnOptional) return TypeName.INT;
+    else if (type == RifColumnType.NUM
+        && columnScale.orElse(Integer.MAX_VALUE) == 0
+        && isColumnOptional) return ClassName.get(Integer.class);
+    else throw new IllegalArgumentException("Unhandled field type: " + type.name());
+  }
+
+  private static TypeName selectJavaPropertyType(
```

Comment:
FWIW, I'd suggest JavaDoc'ing this, as it's not entirely obvious what some of those parameters mean (I can never quite remember what scale means, for instance).

Reply:
added